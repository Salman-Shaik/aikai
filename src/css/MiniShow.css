.mini_poster {
  height: 10vh;
  width: 4.5vw;
  box-shadow: #1d1d1d 1px 1px 8px 8px;
  border-radius: 5px;
}

.mini_show {
  width: 18vw;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  box-shadow: #1d1d1d 1px 1px 8px 8px;
  padding: 5px;
  border-radius: 2px;
  color: papayawhip;
  cursor: pointer;
}

.mini_show_details {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
}

.mini_show_actions {
  height: 10.5vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  padding: 3px;
}

.title {
  margin-left: 1vw;
  margin-top: 1vh;
  background-image: -webkit-linear-gradient(-225deg, #e3fdf5 0%, #ffe6fa 100%);
  background-image: linear-gradient(-225deg, #e3fdf5 0%, #ffe6fa 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  text-align: left;
}
