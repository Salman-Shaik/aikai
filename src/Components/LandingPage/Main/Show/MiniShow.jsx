import Tooltip from "@material-ui/core/Tooltip";
import { CheckCircle } from "@material-ui/icons";
import React from "react";
import "../../../../css/MiniShow.css";
import {
  createCookie,
  imageUrlBuilder,
  onPosterError,
} from "../../../../lib/helper";
import { markWatched } from "../../../../lib/networkCalls";
import { FavoriteComponent } from "./Actions/FavoriteComponent";
import { WatchListComponent } from "./Actions/WatchListComponent";

export const MiniShow = ({
  posterPath,
  title,
  id,
  setHomePageLoaded,
  favFlag,
  watchListFlag,
  updateLocation,
  isUserLoggedIn,
  currentShowType,
}) => {
  const onClick = () => {
    createCookie("showId", id);
    !!currentShowType && createCookie("showType", currentShowType);
    updateLocation("/");
    setHomePageLoaded(false);
  };

  const onWatched = () => markWatched(id, updateLocation);

  return (
    <section
      className="mini_show"
      data-testid={"mini_show"}
      onClick={!!favFlag || !!watchListFlag ? () => {} : onClick}
    >
      <section className="mini_show_details" onClick={onClick}>
        <img
          className="mini_poster"
          data-testid="mini_poster"
          src={imageUrlBuilder(posterPath)}
          alt={title}
          title={title}
          onError={onPosterError}
          onClick={onClick}
        />
        <h4 className="title" onClick={onClick} data-testid="title">
          {title}
        </h4>
      </section>
      <section className="mini_show_actions">
        {!!favFlag && (
          <FavoriteComponent
            id={id}
            title={title}
            posterPath={posterPath}
            initialValue={true}
            updateLocation={updateLocation}
            isUserLoggedIn={isUserLoggedIn}
          />
        )}
        {!!watchListFlag && (
          <WatchListComponent
            id={id}
            title={title}
            posterPath={posterPath}
            updateLocation={updateLocation}
            isUserLoggedIn={isUserLoggedIn}
            initialValue={true}
          />
        )}
        {!!watchListFlag && (
          <Tooltip title="Watched?">
            <CheckCircle className="watched_icon" onClick={onWatched} />
          </Tooltip>
        )}
      </section>
    </section>
  );
};
