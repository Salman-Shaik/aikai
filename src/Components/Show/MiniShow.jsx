import React from "react";
import '../../css/MiniShow.css'
import {imageUrlBuilder} from "../../lib/helper";

export const MiniShow = ({posterPath, title, setCurrentShowId, setHomePageLoaded,id}) => {
    const onClick = () => {
        setHomePageLoaded(false);
        setCurrentShowId(id);
    }

    return <section className="mini_show" onClick={onClick}>
        <img className="mini_poster" src={imageUrlBuilder(posterPath)} alt={title} onClick={onClick}/>
        <h4 className="title" onClick={onClick}>{title}</h4>
    </section>
}