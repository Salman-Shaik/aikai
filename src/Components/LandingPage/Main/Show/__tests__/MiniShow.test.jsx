import { render } from "@testing-library/react";
import React from "react";
import { MiniShow } from "../MiniShow";
describe("Mini Show", () => {
  it("Snapshot Test", function () {
    const { container } = render(<MiniShow />);
    expect(container).toMatchSnapshot();
  });

  it("should render a mini show component", () => {
    const title = "Shazam!";
    const { getByAltText, getByTestId } = render(<MiniShow title={title} />);
    const poster = getByAltText(title);
    const titleComponent = getByTestId("title");
    expect(poster).toBeDefined();
    expect(titleComponent.innerHTML).toBe(title);
  });

  it("should render a mini show component with actions", () => {
    const { getByTestId, queryByTestId } = render(<MiniShow favFlag={true} />);
    const like = getByTestId("favorite");
    const unlike = queryByTestId("unlike");
    expect(like).toBeDefined();
    expect(unlike).toBeNull();
  });
});
