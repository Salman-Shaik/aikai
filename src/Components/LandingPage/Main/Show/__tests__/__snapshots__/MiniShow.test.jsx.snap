// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mini Show Snapshot Test 1`] = `
<div>
  <section
    class="mini_show"
  >
    <section
      class="mini_show_details"
    >
      <img
        class="mini_poster"
        src="https://image.tmdb.org/t/p/originalundefined"
      />
      <h4
        class="title"
        data-testid="title"
      />
    </section>
    <section
      class="mini_show_actions"
    />
  </section>
</div>
`;
